File: template_method/typescript/CajaDeAhorro.ts
class CajaDeAhorro extends CuentaBancaria {

	public constructor(
		unCbu: string, unSaldo: number = 0
	) {
		super(unCbu, unSaldo);
	}

	protected hacerValidar(unMonto: number) {
		return (
			(unMonto > 0) && (unMonto <= this.saldo)
		);
	}

}

File: template_method/python/CajaDeAhorro.py
from CuentaBancaria import CuentaBancaria

class CajaDeAhorro(CuentaBancaria, object):

	def __init__(self, unCbu, unSaldo):
		super(unCbu, unSaldo)

	def hacerValidar(self, unMonto):
		return (unMonto > 0) and (unMonto <= self.__saldo)